packages/zero-client/src/client/query-manager.ts

```diff
@@ -17,14 +20,19 @@ import {
 import type {TableSchema} from '../../../zero-schema/src/table-schema.ts';
 import type {GotCallback} from '../../../zql/src/query/query-impl.ts';
 import {compareTTL, parseTTL, type TTL} from '../../../zql/src/query/ttl.ts';
+import type {ReadonlyJSONValue} from '../../../shared/src/json.ts';
 import {desiredQueriesPrefixForClient, GOT_QUERIES_KEY_PREFIX} from './keys.ts';
 import type {MutationTracker} from './mutation-tracker.ts';
 import type {ReadTransaction} from './replicache-types.ts';
 
 type QueryHash = string;
 
 type Entry = {
-  normalized: AST;
+  // May be undefined if the entry is a custom (named) query.
```

Comment:
I think a discriminated union type would be better here (then you wouldn't need the comments in various places about fields being undefined for custom vs adhoc, it would be enforced by the type), but don't feel strongly about it since the 'ad-hoc' query type is going to go away.